src/ccache.cpp

```diff
@@ -728,13 +728,55 @@ result_name_from_depfile(Context& ctx, struct hash* hash)
   return d;
 }
 
+// Execute the compiler/preprocessor, with logic to retry without requesting
+// colored diagnostics messages if that fails.
+static int
+execute(Context& ctx,
+        Args& args,
+        const std::string& stdout_path,
+        int stdout_fd,
+        const std::string& stderr_path,
+        int stderr_fd)
+{
+  if (ctx.diagnostics_color_failed) {
+    if (ctx.guessed_compiler == GuessedCompiler::gcc) {
+      args.erase_with_prefix("-fdiagnostics-color");
+    }
+  }
```

Comment:
The reason I didn't do it this way was in case another compiler-specific block needs to be added in the future. Then it would be just added lines in the diff.

Reply:
Then we have different taste.

Reply:
Hmm, I wouldn't say it's a matter of taste, as this is not a style-driven consideration. Merge conflicts between future additions at this location will be less likely to occur (and more straightforward to resolve if they do occur) if they don't need to touch the outer `if` statement on line 741. But, :shrug: it's not my codebase, so whatever.